### ISSUE TITLE
Separate redux selector definition from reducers & action creator definitions

### ISSUE THREAD

**Issue by 0xDaedalus** [tallyhowallet/extension#1507]
We have a smattering of dangling selectors defined inside of `redux-slices/____.ts` files.  We should port them over to `redux-slices/selectors/____.ts` files where appropriate and define new files inside of `redux-slices/selectors` in the cases where we do not have a corresponding `selectors` file.

**Comment by greg-nagy** [comment]
I am not strictly against this idea, but I have lived a transition in angular land where the structure went according to type and not functionality — so there were `services` folder and `components` folder etc. This idea gave me the same idea. 

The problem with that approach was that as it turned out, usually when I am looking for something I am doing it in the context of the given feature/functionality. So "Where is XYZ selector" was a rare thought. It was more "I am working on this UI part, and I need the selector for this".

Long story short the end result in angular land was to organize everything based on functionality and not by type.

But this is just one approach. I have lived this transition and it stuck with me. So I wanted to leave my 2c here with you.

**Comment by 0xDaedalus** [comment]
This has been completed ✅
